File: LivrusVirtus/FoxToolKit/FXPython/ConLivros.py
from FXPy.fox import *
from Consultas import frmConsultas
from resource import Resource
from ConsultasSQL import ConsultasSQL
from RotinasGlobais import RotinasGlobais
import Principal

class frmConLivros(frmConsultas):

  def __init__(self, owner):
    frmConsultas.__init__(self, owner, Resource.STR_CONLIVRO)  

    FXMAPFUNC(self, SEL_COMMAND, frmConLivros.ID_FRMCONSULTAS_BTNPESQUISAR, 
      frmConLivros.onBtnPesquisar)
    FXMAPFUNC(self, SEL_COMMAND, frmConLivros.ID_FRMCONSULTAS_BTNLIMPAR, 
      frmConLivros.onBtnLimpar)
    FXMAPFUNC(self, SEL_SELECTED, frmConLivros.ID_FRMCONSULTAS_GRIDCONSULTA, 
      frmConLivros.onGridSelected)
    FXMAPFUNC(self, SEL_CHANGED, frmConLivros.ID_FRMCONSULTAS_GRIDCONSULTA, 
      frmConLivros.onGridChanged)
    
    self.Rotinas = RotinasGlobais.Rotinas
    self.labDescricao.setText("Livro:")
    self.Rotinas.IniciaLayoutGrid(self.gridConsulta, 2, 13)
    self.gridConsulta.setItemText(0, 1, "ISBN")
    self.gridConsulta.setColumnWidth(1, 80)
    self.gridConsulta.setItemText(0, 2, "Título")
    self.gridConsulta.setColumnWidth(2, 180)
  
    self.nLargura = ( 80,90,80,80,100,80,100,80,80,80 )
    self.sRotulos = ( "Edi\xe7\xe3o", "AnoPublica\xe7\xe3o", "Volume",
      "CodEditora", "Editora", "CodIdioma", "Idioma",
      "NumP\xe1ginas", "Pre\xe7o", "Estoque" )
    self.Rotinas.AdicionaColunasGrid(self.gridConsulta, 3, 10, 
      self.nLargura, self.sRotulos)

  def create(self):
    frmConsultas.create(self)
  
  def onBtnPesquisar(self, sender, sel, ptr):
    self.PesquisarDados(ConsultasSQL.ConsSQL.Livro('S',"", 
      self.edtDescricao.getText(),"","","","","","","",""), 
      Resource.STR_LIVNENC, 12)
    return True
  
  def onBtnLimpar(self, sender, sel, ptr):
    self.LimparDados()
    return True
  
  def onGridSelected(self, sender, sel, ptr):
    self.setLabRegistros(sender.getCurrentRow())
    return True
  
  def onGridChanged(self, sender, sel, ptr):
    self.setLabRegistros(sender.getCurrentRow())
    return True

File: LivrusVirtus/FoxToolKit/FXRuby/ConLivros.rb
require 'fox'
require 'fox/responder'
require 'resource'
require 'Principal'
require 'ConsultasSQL'
require 'Consultas'
include Fox
include Responder

class FRMConLivros < FRMConsultas
  
  def initialize(owner)  
    super(owner, Resource::STR_CONLIVRO)

    FXMAPFUNC(SEL_COMMAND, ID_FRMCONSULTAS_BTNPESQUISAR, :onBtnPesquisar)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCONSULTAS_BTNLIMPAR, :onBtnLimpar)
    FXMAPFUNC(SEL_SELECTED, ID_FRMCONSULTAS_GRIDCONSULTA, :onGridSelected)
    FXMAPFUNC(SEL_CHANGED, ID_FRMCONSULTAS_GRIDCONSULTA, :onGridChanged)
    
    @labDescricao.setText("Livro:")
    $Rotinas.IniciaLayoutGrid(@gridConsulta, 2, 13)
    @gridConsulta.setItemText(0, 1, "ISBN")
    @gridConsulta.setColumnWidth(1, 80)
    @gridConsulta.setItemText(0, 2, "Título")
    @gridConsulta.setColumnWidth(2, 180)
  
    nLargura = [ 80,90,80,80,100,80,100,80,80,80 ]
    sRotulos = [ "Edição", "AnoPublicação", "Volume",
      "CodEditora", "Editora", "CodIdioma", "Idioma",
      "NumPáginas", "Preço", "Estoque" ]
    $Rotinas.AdicionaColunasGrid(@gridConsulta, 3, 10, nLargura, sRotulos)
  end

  def create()
    super
    show(PLACEMENT_SCREEN)
  end
  
  def onBtnPesquisar(sender, sel, ptr)
    PesquisarDados($ConsSQL.Livro('S',"", @edtDescricao.getText(),
      "","","","","","","",""), Resource::STR_LIVNENC, 12)
    return true
  end
  
  def onBtnLimpar(sender, sel, ptr)
    LimparDados()
    return true
  end
  
  def onGridSelected(sender, sel, ptr)
    setLabRegistros(ptr.fm.row)
    return true
  end
  
  def onGridChanged(sender, sel, ptr)
    setLabRegistros(ptr.row)
    return true
  end

end
